ui/tests/integration/components/variable-form-test.js

```diff
@@ -280,15 +280,52 @@ module('Integration | Component | variable-form', function (hooks) {
         })
       );
 
-      await render(hbs`<VariableForm @model={{this.mockedModel}} />`);
-
-      await typeIn('.key-value label:nth-child(1) input', 'superSecret');
-      assert.dom('.key-value-error').doesNotExist();
-
-      find('.key-value label:nth-child(1) input').value = '';
-
-      await typeIn('.key-value label:nth-child(1) input', 'super.secret');
-      assert.dom('.key-value-error').exists();
+      const testCases = [
+        {
+          name: 'valid key',
+          key: 'superSecret2',
+          warn: false,
+        },
+        {
+          name: 'invalid key with dot',
+          key: 'super.secret',
+          warn: true,
+        },
+        {
+          name: 'invalid key with slash',
+          key: 'super/secret',
+          warn: true,
+        },
+        {
+          name: 'invalid key with emoji',
+          key: 'supersecretspy🕵️',
+          warn: true,
+        },
+        {
+          name: 'unicode letters',
+          key: '世界',
+          warn: false,
+        },
+        {
+          name: 'unicode numbers',
+          key: '٣٢١',
+          warn: false,
+        },
+        {
+          name: 'unicode letters and numbers',
+          key: '世٢界١',
+          warn: false,
+        },
+      ];
+      for (const tc of testCases) {
+        await render(hbs`<VariableForm @model={{this.mockedModel}} />`);
+        await typeIn('.key-value label:nth-child(1) input', tc.key);
```

Comment:
`nit`: If we edit the order of our markup this test will fail. It would be nice to use a `data-test` attribute here. However, it's not necessary.

Reply:
Good point, fixed in https://github.com/hashicorp/nomad/pull/15933/commits/8800e0810b88a327edc6d12324d716cef04f2fc6 👍